source/blog/2017-07-31-fedora-atomic-25-to-26-upgrade.md

```diff
@@ -0,0 +1,274 @@
+                Version: 26.91 (2017-07-23 03:12:08)
+                 Commit: 0715ce81064c30d34ed52ef811a3ad5e5d6a34da980bf35b19312489b32d9b83
+           GPGSignature: 1 signature
+                         Signature made Sun 23 Jul 2017 03:13:09 AM UTC using RSA key ID 812A6B4B64DAB85D
+                         Good signature from "Fedora 26 Primary <fedora-26-primary@fedoraproject.org>"
+
+  fedora-atomic:fedora-atomic/25/x86_64/docker-host
+                Version: 25.154 (2017-07-04 01:38:10)
+                 Commit: ce555fa89da934e6eef23764fb40e8333234b8b60b6f688222247c958e5ebd5b
+[vagrant@host ~]$ cat /etc/fedora-release 
+Fedora release 26 (Twenty Six)
+```
+
+We are now on Fedora 26 Atomic Host. Now is a good time to check your
+services (most likely running in containers) to see if they are still
+working. If not, then you always have the rollback command: `sudo
+rpm-ostree rollback`.
+
+**NOTE:** Over time you can see updated commands for upgrading Atomic
+          Host between releases by visiting [this](https://fedoraproject.org/wiki/Atomic_Host_upgrade)
+          wiki page.
+
+# Appendix A: Upgrading Systems with Kubernetes
+
+Fedora 25 Atomic Host ships with Kubernetes **v1.5.3**, and Fedora 26
+Atomic Host ships with Kubernetes **v1.6.7**. **Before** upgrading systems
+participating in an existing Kubernetes cluster from 25 to 26, there
+are a few configuration changes to make.
+
+## Node Servers
+
+In Kubernetes 1.6, the `--config` argument is no longer valid. If
+your `KUBELET_ARGS` in `/etc/kubernetes/kubelet` point to the manifests
+directory using the `--config` argument, then you need to change
+the argument name to `--pod-manifest-path`. Also in `KUBELET_ARGS`, you 
+need to add the argument `--cgroup-driver=systemd`.
+
+For example, if your `/etc/kubernetes/kubelet` file starts with the
+following:
+
+```nohighlight
+KUBELET_ARGS="--kubeconfig=/etc/kubernetes/kubelet.kubeconfig --config=/etc/kubernetes/manifests --cluster-dns=10.254.0.10 --cluster-domain=cluster.local"
+```
+
+Then it should be changed to be:
+
+```nohighlight
+KUBELET_ARGS="--kubeconfig=/etc/kubernetes/kubelet.kubeconfig --pod-manifest-path=/etc/kubernetes/manifests --cluster-dns=10.254.0.10 --cluster-domain=cluster.local --cgroup-driver=systemd"
+```
+
+## Master Servers
+
+### Staying With etcd2 
+
+From Kubernetes 1.5 to 1.6 upstream 
+[shifted](https://kubernetes.io/docs/tasks/administer-cluster/upgrade-1-6/)
+from using version 2 of the etcd API to version 3. The 
+[Kubernetes documentation](https://github.com/kubernetes/kubernetes/blob/93b144c/CHANGELOG.md#internal-storage-layer-1)
+instructs users to **add** two arguments to the `KUBE_API_ARGS` variable
+in the `/etc/kubernetes/apiserver` file:
+
+```nohighlight
+--storage-backend=etcd2 --storage-media-type=application/json
+```
+
+This will ensure that any pods, services or other objects stored in etcd
+will continue to be found by Kubernetes once you've completed your upgrade.
```

Comment:
Note that the kubernetes project will soon deprecate support for etcd2 (it's already not well-covered in tests).  However, since the deprecation hasn't been officially announced, I guess we don't need to mention it.